analyzers/src/SonarAnalyzer.CSharp/Rules/XXE/XmlReaderSettingsValidator.cs

```diff
@@ -111,16 +114,17 @@ public IList<Location> GetUnsafeAssignmentLocations(InvocationExpressionSyntax i
         private static bool IsXmlReaderSettings(ExpressionSyntax expressionSyntax, SemanticModel semanticModel) =>
             semanticModel.GetTypeInfo(expressionSyntax).Type.Is(KnownType.System_Xml_XmlReaderSettings);
 
-        private static ObjectCreationExpressionSyntax GetObjectCreation(ISymbol symbol, InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
+        private static IObjectCreation GetObjectCreation(ISymbol symbol, InvocationExpressionSyntax invocation, SemanticModel semanticModel) =>
             // First we search for object creations at the syntax level to see if the object is created inline
             // and if not we look for the identifier declaration.
             invocation.DescendantNodes()
                       .Union(symbol.GetLocationNodes(invocation))
-                      .OfType<ObjectCreationExpressionSyntax>()
+                      .Where(x => x.IsAnyKind(SyntaxKind.ObjectCreationExpression, SyntaxKindEx.ImplicitObjectCreationExpression))
+                      .Select(ObjectCreationFactory.Create)
                       .FirstOrDefault(objectCreation => IsXmlReaderSettingsCreationWithInitializer(objectCreation, semanticModel));
 
-        private static bool IsXmlReaderSettingsCreationWithInitializer(ObjectCreationExpressionSyntax objectCreation, SemanticModel semanticModel) =>
-            objectCreation.Initializer != null && IsXmlReaderSettings(objectCreation, semanticModel);
+        private static bool IsXmlReaderSettingsCreationWithInitializer(IObjectCreation objectCreation, SemanticModel semanticModel) =>
```

Comment:
Method `IsXmlReaderSettings` on line 114 is now used only once and can be embedded into it's call site instead.

Reply:
_Just to be clear:_ I want to preserve `IsXmlReaderSettingsCreationWithInitializer` as is, because it's used on a long line in nested lambda and the name brings some readability as well.